examples/tutorials/advanced/focal_mechanisms.py

```diff
@@ -0,0 +1,359 @@
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=aki_dict_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    # Use a 1-point thick, darkorange and solid line
+    pen="1p,darkorange,solid",
+)
+
+fig.meca(
+    spec=aki_dict_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    outline="1p,darkorange,solid",
+)
+
+fig.show()
+
+
+# %%
+# Highlighting the nodal planes
+# -----------------------------
+#
+# Use the parameter ``nodal``, whereby ``"0"`` refers to both, ``"1"`` to the first,
+# and ``"2"`` to the second nodal plane(s). Only the circumference and the specified
+# nodal plane(s) are plotted, i.e. the quadrants remain unfilled (transparent). If
+# needed, make usage of the stacking concept of (Py)GMT and use ``nodal`` with the
+# ``outline`` or / and ``pen``  parameters in combination.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=aki_dict_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    nodal="0/1p,black,solid",
+)
+
+fig.meca(
+    spec=aki_dict_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="lightorange",
+    outline="0.5p,black,solid",
+)
+fig.meca(
+    spec=aki_dict_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    nodal="1/1p,darkorange,solid",
+)
+fig.meca(
+    spec=aki_dict_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="white@100",
+    extensionfill="white@100",
+    pen="1p,gray30,solid",
+)
+fig.show()
+
+
+# %%
+# Adding offset from event location
+# ---------------------------------
+#
+# Specify the optional parameters ``plot_longitude`` and ``plot_latitude``.
+# Additionally the parameter ``offset`` has to be set. Besides just drawing a line
+# between the beachball and the event location, a small circle can be plotted at the
+# event location by appending **+s** and the descired circle diameter. The connecting
+# line as well as the outline of the circle are plotted with the  setting of pen, or
+# can be adjusted separately. The fill of the small circle corresponds to the fill
+# of the compressive quadrantes.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=aki_dict_single,
+    scale="1c",
+    longitude=-1,
+    latitude=0,
+    depth=0,
+    plot_longitude=-3,
+    plot_latitude=2,
+    offset=True,
+)
+
+fig.meca(
+    spec=aki_dict_single,
+    scale="1c",
+    longitude=3,
+    latitude=0,
+    depth=0,
+    plot_longitude=1,
+    plot_latitude=2,
+    offset="+p1p,darkorange+s0.25c",
+    compressionfill="lightorange",
+)
+
+fig.show()
+
+
+# %%
+# Plotting multiple beachballs
+# ----------------------------
+#
+# Data of four earthquakes taken from USGS.
+# Provide lists.
```

Comment:
Need changes.

Reply:
Oh, yeah! Added in commit https://github.com/GenericMappingTools/pygmt/pull/2550/commits/379cdb1137ce6d43b89e42408d870fabf9bed357.